sallyport/src/lib.rs

```diff
@@ -171,6 +171,49 @@ impl Block {
     pub const fn buf_capacity() -> usize {
         Page::size() - size_of::<Message>()
     }
+
+    /// Returns a Cursor for the Block
+    pub fn cursor<'a>(&'a mut self) -> Cursor<'a> {
+        Cursor(&mut self.buf)
+    }
+}
+
+/// Helper for allocation of untrusted memory in a Block.
+pub struct Cursor<'a>(&'a mut [u8]);
+
+impl<'a> Cursor<'a> {
+    /// Allocates an array, containing count number of T items. The result is uninitialized.
+    ///
+    /// # Safety
+    ///
+    /// This function is unsafe because it returns an uninitialized array.
+    pub unsafe fn alloc<T>(&'a mut self, count: usize) -> Result<&'a mut [T], libc::c_int> {
```

Comment:
```suggestion
    pub unsafe fn alloc<T>(&'a mut self, count: usize) -> Result<&'a mut [T], ()> {
```

Reply:
Callers may want different errors in different contexts. So we'll just let the caller do `Result::or()`.